sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java

```diff
@@ -124,10 +137,14 @@ public FileBasedSource(String fileName, long minBundleSize,
       long startOffset, long endOffset) {
     super(startOffset, endOffset, minBundleSize);
     mode = Mode.SINGLE_FILE_OR_SUBRANGE;
-    this.fileOrPatternSpec = fileName;
+    this.fileOrPatternSpec = StaticValueProvider.of(fileName);
   }
 
   public final String getFileOrPatternSpec() {
```

Comment:
Was keeping this around for the time being to avoid changing a bunch of callers (Avro, XML, etc.) in this PR.